### PR TITLE
feat: README samples (again)

### PR DESCRIPTION
**Pull request by mbonig**
Re-adds the README file and sample file rework. Now uses the synth and _outdir instead of __dirname.

### CODE REVIEW

File: src/sample-file.ts [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+   * @param dir directory to add files to. If directory already exists, nothing is added.
+   * @param options options for which files to create.
+   */
+  constructor(project: Project, dir: string, options: SampleDirOptions) {
+    super(project);
+    this.dir = dir;
+    this.options = options;
+  }
+
+  synthesize(_outdir: string) {
```
Comment (Chriscbr) [inline src/sample-file.ts:88]:
```suggestion
  public synthesize(outdir: string) {
```

**Comment by mbonig** [comment]
do y'all care if I squash my commits?

File: src/sample-file.ts [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+   * @param contents - the contents of the file to write
+   * @return boolean - whether a new file was written or not.
+   * @private
+   */
+  private writeOnceFileContents(dir: string, filename: string, contents: string) {
+    const fullFilename = path.join(dir, filename);
+    if (fs.existsSync(fullFilename)) {
+      return;
+    }
+    fs.mkdirpSync(dir);
```
Comment (eladb) [inline src/sample-file.ts:54]: please use `writeFile` from `util.ts` which takes care of creating the directory tree

File: src/sample-file.ts [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+
+  public synthesize(outdir: string) {
+    const fullOutdir = path.join(outdir, this.dir);
+    if (fs.pathExistsSync(fullOutdir)) {
+      return;
+    }
+
+    fs.mkdirpSync(fullOutdir);
+    for (const filename in this.options.files) {
+      fs.writeFileSync(path.join(fullOutdir, filename), this.options.files[filename]);
```
Comment (eladb) [inline src/sample-file.ts:96]: Please use writeFile from util.ts (and then mkdirp is not required)

**Review by eladb — CHANGES_REQUESTED** [review]
Can you please confirm that you have manually tested that `projen new` works?

**Comment by mbonig** [comment]
> Can you please confirm that you have manually tested that `projen new` works?

Yes. I went into `test/` and created a new directory, cd'd into that directory and then did `../../bin/projen new awscdk-app-ts`. The resulting code built as expected (projen runs without errors and a `yarn cdk synth` compiles without errors).

File: src/sample-file.ts [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+   * @param contents - the contents of the file to write
+   * @return boolean - whether a new file was written or not.
+   * @private
+   */
+  private writeOnceFileContents(dir: string, filename: string, contents: string) {
+    const fullFilename = path.join(dir, filename);
+    if (fs.existsSync(fullFilename)) {
+      return;
+    }
+    writeFile(fullFilename, contents);
```
Comment (eladb) [inline src/sample-file.ts:55]: By default I think `writeFile` creates readonly files, so here you'd need to explicitly set:

```ts
writeFile(fullFilename, contents, { readonly: false });
```

**Review by eladb — CHANGES_REQUESTED** [review]
files are written as read-only.

**Review by eladb — APPROVED** [review]

**Pull request merged**